### PR TITLE
March into Mapness - Traveller's Rest (Space structure)

### PR DESCRIPTION
**Pull request by LordVollkorn**
## About The Pull Request

I thought it would be fun to learn mapping with [this guide](https://hackmd.io/QBaz0CaTRhOp7pEnUlXnzQ), and perhaps along the way I could make a bit of content that I would like to see in space some day.

I wanted to create an only recently abandoned space structure that is relatively small but could be used as a pitstop for space-travelling spessmen to rest and recharge, perhaps even be used as a second base for further exploration.

The structure starts unpowered, but after you break in, you can turn on the APC (starts unlocked) and set the solar panels to generate power. With all it's amenities, it's similar to the "Space Bunker" (deepstorage.dmm), but of course much smaller. There is not much in terms of loot, just a random toolbox, night vision goggles and a better capacitator for the charger.

With all the new space ruins upcoming, perhaps its not the worst idea to have a place like this.

grafik
grafik

_Since this is my first map, I might have missed some things.
I am a little bit torn between "it does not look abandonded enough" and "it should look nice because its a resting point"._

### Mapping March

Ckey to recieve rewards: LordVollkorn

## Changelog
:cl: LordVollkorn / Gage Gaebler
add: Added Traveler's Rest, a tiny space structure for space travellers to relax and to prepare for the explorations ahead.
/:cl:

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
What's the point of the four 1x1's with windoors?

**Comment by ZephyrTFA** [comment]
wrong button

File: code/datums/ruins/space.dm [hunk 1]
```diff
@@ -350,3 +350,9 @@
 	suffix = "dangerous_research.dmm"
 	name = "Alternate Sciences Research Center"
 	description = "When you're messing with the occult, who knows what you're going to get?"
+
+/datum/map_template/ruin/space/travellersrest
```
Comment (Mothblocks) [inline code/datums/ruins/space.dm:354]: `travellers_rest`

**Comment by LordVollkorn** [comment]
> What's the point of the four 1x1's with windoors?

I really liked the light arrangement and the lights were originally encased in a full glass tile, but then I noticed lights were breaking and they couldnt be reached without deconstruction.
So I figured with windoors I could keep the looks, still reach the lights, and perhaps gain the one tile as storage space.
(Might sound strange but I couldnt find a way to make the lights reachable and make it look nice)

File: code/datums/ruins/space.dm (hunk 1 above)
Comment (LordVollkorn) [reply code/datums/ruins/space.dm:354]: just in this one spot or for all instances of this word?

Comment (Mothblocks) [reply code/datums/ruins/space.dm:354]: All preferably

**Comment by LemonInTheDark** [comment]
what does it look like with starlight? (uncomment the STARLIGHT config in game_options.txt)

File: code/datums/ruins/space.dm (hunk 1 above)
Comment (LordVollkorn) [reply code/datums/ruins/space.dm]: All changed

**Comment by LordVollkorn** [comment]
> what does it look like with starlight? (uncomment the STARLIGHT config in game_options.txt)

Unsure if there is a visual difference from the screenshots from the description above

grafik

grafik
grafik

**Comment by Mothblocks** [comment]
You don't have to click the update branch button, it is not an error

**Comment by Mothblocks** [comment]
image

**Comment by MMMiracles** [comment]
I like the general layout and aesthetic, especially for a first-time map. I only really have some suggestions apart from two minor mapping issues.

Issues:
- Circled solar arrays have doubled up panels.
image
- The area next to the grinder is using the ruin's area instead of the area passthrough.

Suggestions:
- Maybe add some lattices outside with some basic debris protection (i.e grilles with girders under them)? I don't believe space debris is actually an issue outside of the station z-level but thematically it seems weird to have something so fragile with zero debris catchers.
- Maybe replace the large lights with fewer/small bulb ones? That APC is probably gonna be chugging hard with 8 large lights on top of all the other existing equipment on that grid, especially since it has no SMES to store excess power and I assume that APC is on a standard cell by default.
- I would suggest at least adding a spot for a spare oxygen tank and a mask. I would of also said even a super simple atmos piping (an air tank for distro and waste pumping into space) but the ruin itself is pretty small and having to add one would mess up the current layout.

Otherwise looks cool. Would get drunk in if I fell out the escape shuttle and flew into this.

**Comment by private-tristan** [comment]
Do holopads work off Zlevel? If so it could be a unique way for space explorers to contact the station.

**Comment by VladinXXV** [comment]
toilet right next to the window. funny. bold?
image

**Comment by coiax** [comment]
My main concerns are about long term thematic/mechanical viability:

APCs generally have small batteries, are the solar panels enough to sustain the base at full power?
With no air supply, there'd be an upper limit to how long you could stay in this place. I doubt you'd reach it in normal gameplay though.

Could add an air cannister or a portable scrubber to acknowledge the atmospheric situation, an adhoc atmos setup.

I suppose thematically you could say the potted plants consume CO2, but also they're like immortal unchangable plants that require no water.

(Also, I'll get yelled at by the mapping crew, but I do not like tiny fans to "keep the vaccum" out. But I don't think we really have a proper airlock chamber setup that's easily usable, so not something that can be solved quickly)

---

But also none of these concerns are enough to stop this cute little ruin being added. 🙇

**Comment by LordVollkorn** [comment]
@MMMiracles Wow, thanks! I didn't even notice the solar arrays were double; Is there a way of checking these things without right-clicking the tiles?
I have fixed the arrays and the space tiles accordingly 👍

About the lattice: I placed some around the structure before, but that made it a bit too big and the view out the window wasn't so nice. If possible I would like to not use the lattice at all (and like you mentioned, I also think that space debris is not generated in deep space)

Good point with the lights and the power: Before i made this PR, I actually used an SMES but I noticed that with the current setup it was not neccessary; The power levels always stay at good levels, even with all lights burning and some machinery usage.

I don't know if its possible to integrate an atmos system in such a small space, but the only related thing it has is a tiny fan in the outer airlock. The tank and internals are a good idea though; I added some to the entrace area and moved the pAI to an empty table.
grafik

**Comment by LordVollkorn** [comment]
> Do holopads work off Zlevel? If so it could be a unique way for space explorers to contact the station.
@private-tristan 
Good question, I originally added it because I wanted to have a pre-recorded hologram message of the original owner of the structure, but in the end I just left it in for the aesthetics.
Might need to pick that original idea back up, if I can find some example I can use from another map

**Comment by LordVollkorn** [comment]
@coiax 
Good points! I have tested the structure for 20-30 minutes in a round and as far as I can tell, the APC with the solar panels attached can handle things pretty good. There is a tiny fan in the outer airlock to prevent air from escaping, but only because small mistakes (like dragging a big object behind you) could drain the structure really quickly (and I dont have enough space for an atmos setup).
 I added an O2 tank and a mask to the entrance area now, just to make sure!

I didn't give too much thought to the plants, they might be made of plastic :D

**Comment by LordVollkorn** [comment]
Up-to-date screenshot with all changes mentioned before
grafik

**Comment by newfren** [comment]
a booze dispenser without accompanying soda dispenser really limits the drinks that can actually be made with it

**Comment by LordVollkorn** [comment]
> a booze dispenser without accompanying soda dispenser really limits the drinks that can actually be made with it
@newfren 
I might be able squeeze one in somehow, let me see what I can do

**Comment by ArthuroPep** [comment]
I want to go there with a communication relay then spend the remainder of the shift talking on the radio and rping as Esker from Outer Wilds, greeting travelers among the way

**Comment by LordVollkorn** [comment]
> I want to go there with a communication relay then spend the remainder of the shift talking on the radio and rping as Esker from Outer Wilds, greeting travelers among the way

@ArthuroPep
I like this idea, and I think those 4 compartments where the lights are in can be used to house the machines

**Comment by LordVollkorn** [comment]
I managed to squeeze the last few pixels out of the bar area
grafik

And since it was easy to add, I also included a light switch and a newscaster
grafik

**Comment by LordVollkorn** [comment]
grafik

File: code/datums/holocall.dm [hunk 2]
```diff
@@ -476,3 +476,25 @@
...
+	DELAY 50
+	SAY You can crash here if you need to, but make sure to check the anchor cables before you leave.
+	DELAY 50
+	SAY If you don't, this thing might drift off into space.
+	DELAY 50
+	SAY Then some weirdo could find it and potentially claim it as their own.
+	DELAY 50
+	SAY Anyway, gotta go, see ya!
+	DELAY 40
+	SOUND sparks"}
```
Comment (Mothblocks) [inline code/datums/holocall.dm:500]: Indent these all up by one, and have `"}` on its own line (dedented one)

File: code/game/area/areas/ruins/space.dm [hunk 3]
```diff
@@ -570,3 +570,7 @@
 /area/ruin/space/has_grav/powered/clownplanet
 	name = "\improper Clown Planet"
 	ambientsounds = list('sound/ambience/clown.ogg')
+
+// The abandonded capsule 'The Travellers Rest'
```
Comment (Mothblocks) [inline code/game/area/areas/ruins/space.dm:574]:
```suggestion
// The abandoned capsule 'The Traveller's Rest'
```

**Comment by Mothblocks** [comment]
I also just realized it's spelled "Traveler's" and not "Traveller's" if you care enough

File: code/datums/holocall.dm (hunk 2 above)
Comment (LordVollkorn) [reply code/datums/holocall.dm:500]: @Mothblocks Yes sir

**Comment by LordVollkorn** [comment]
> I also just realized it's spelled "Traveler's" and not "Traveller's" if you care enough

@Mothblocks 
I had to google it and you are right
grafik

I should be able to change that

**Comment by Mothblocks** [comment]
Oh if it's a British thing then you didn't need to care (we explicitly don't have a preference between US and British english), I didn't know that was a regional thing

**Comment by LordVollkorn** [comment]
All good, we can go with the US version, I have no objections either way 👍

**Comment by LordVollkorn** [comment]
Tiny "atmos system"
grafik

**Comment by LordVollkorn** [comment]
I feel pretty good about this map now, and with all the feedback and fixed issues I would say that this is ready to go.

MapDiffBot2 seems to error out, but what I see from other PR's this seems to be non-critical?

**Comment by Zytolg** [comment]
For the Atmos System, could you reverse the tank to DIR_8 [WEST] and route the pipes down to a default on_state pump with 108 pressure? The way atmos is set up on this is going to result in your pressure tank draining to the capacity of the pipes (4500psi?) in an effort to reach pressure equilibrium and making any player-facing modifications to this ruin hazardous at best (they'll be blown away from unbranching pipes). You'll also want to set a waste loop for air on another pipe network and at least hook up a passive vent into space for it. Otherwise you'll just be storing air inside the scrubbers network until it hits that pipe psi cap and cease functioning.

**Comment by LordVollkorn** [comment]
> For the Atmos System, could you reverse the tank to DIR_8 [WEST] and route the pipes down to a default on_state pump with 108 pressure? The way atmos is set up on this is going to result in your pressure tank draining to the capacity of the pipes (4500psi?) in an effort to reach pressure equilibrium and making any player-facing modifications to this ruin hazardous at best (they'll be blown away from unbranching pipes). You'll also want to set a waste loop for air on another pipe network and at least hook up a passive vent into space for it. Otherwise you'll just be storing air inside the scrubbers network until it hits that pipe psi cap and cease functioning.

@Zytolg
Hi, you good some good points there.
For the tank: I attached the pump like you said, I didn't even think about it until you mentioned it. But I opted for facing it south, because its easier to reach when under the table (in my opinion).
Or is there a specific reason to go west first?

I put the red pipes in layer 2 like on the normal station and I placed a passive vent outside (before I used an injector, it was under the solar panel)

**Comment by coiax** [comment]
Much as I admire your efforts in cramming in the machinery into the cramped bar system, is it possible to replicate that setup if you were constructing it only ingame? That's what I'm concerned about with the pixel shifted microwaves and chemical setups.

(Note I'm not saying "oh you should totally modify the whole construction system to make this possible", it's more I have a natural dislike of any mapping setup that could not be constructed in-universe.

You could move the air tank to be on the outside, rather than inside, and if you put another table on the turf with the air alarm, then you could store the beakers and damp rag on the corner table. 🤔

**Comment by LordVollkorn** [comment]
> Much as I admire your efforts in cramming in the machinery into the cramped bar system, is it possible to replicate that setup if you were constructing it only ingame? That's what I'm concerned about with the pixel shifted microwaves and chemical setups.
> 
> (Note I'm not saying "oh you should totally modify the whole construction system to make this possible", it's more I have a natural dislike of any mapping setup that could not be constructed in-universe.)
> 
> You could move the air tank to be on the outside, rather than inside, and if you put another table on the turf with the air alarm, then you could store the beakers and damp rag on the corner table. 🤔

@coiax 
I get what you mean; Basically its not possible (as far as I know) to replicate the bar or chemical setup with ingame tools. If this map was a fully fledged station, then of course I shouldn't work with pixel shifts like this.

But just in this case, for a unique, tiny space structure, I would like to keep the design as a "gimmicky tinyhome in space" if that makes sense 😁

**Comment by Addust** [comment]
oh my god yes

**Comment by VladinXXV** [comment]
> But just in this case, for a unique, tiny space structure, I would like to keep the design as a "gimmicky tinyhome in space" if that makes sense 😁

image

You could de-pixel shift the microwave and move the All-In-One Grinder to the table where the box of drinking glasses and rag are. This keeps them all roughly in the same area while allowing it to be rebuilt ingame more accurately.

image

Also, the tile underneath the toil is missing the texture dot things every other tile in that section has. Intentional?

**Comment by VladinXXV** [comment]
image

I only just noticed the chem setup, too. I strongly advise against this. I would move the pAI and bottle of space cleaner onto the same table as the box of beakers, remove the table they were previously on, put the reaction chamber in that tile, and de-pixel shift all machines.

**Comment by LordVollkorn** [comment]
@coiax @VladinXXV 
Although I liked the crammed design, I will take your feedback and make the thing in-game possible to rebuild.
I have de-pixelshifted the bar and chemistry, and there is only ever 1 machine per tile.
(Also corrected the toilet tile, I never noticed!)

**Comment by LordVollkorn** [comment]
Newest version:
grafik
grafik

**Comment by san7890** [comment]
you must add this to https://github.com/tgstation/tgstation/blob/master/config/spaceruinblacklist.txt

**Comment by LordVollkorn** [comment]
> you must add this to https://github.com/tgstation/tgstation/blob/master/config/spaceruinblacklist.txt

@san7890 Yessir, added the line

**Review by san7890 — COMMENTED** [review]
i do not believe that the walls to this structure have to be reinforced... there's not really much reason for a space ruin to be that secure? if you think there's value to it feel free to keep it but I don't really see much of any

**Comment by LordVollkorn** [comment]
> i do not believe that the walls to this structure have to be reinforced... there's not really much reason for a space ruin to be that secure? if you think there's value to it feel free to keep it but I don't really see much of any

@san7890 I have taken your suggestion and replaced the walls; You were right: There was no particular reason for the walls to be reinforced, other than the looks. But for me normal walls work absolutely fine as well.

**Comment by Rex9001** [comment]
Cool ruin! We 100% need more of these rest type areas

**Pull request merged**